tst/scripts/utils/artemis.py

```diff
@@ -96,18 +119,19 @@ def make(cmake_args, make_nproc):
 def run(nproc, input_filename, arguments, restart=None):
     # global run_directory
     out_log = LogPipe("artemis.run", logging.INFO)
-    os.makedirs(artemis_run_dir, exist_ok=True)
 
     # Build the run command
-    run_command = ["mpiexec", "--oversubscribe", "-n", str(nproc), artemis_executable]
+    run_command = ["mpiexec"]
+    if use_mpi_oversubscribe:
+        run_command += ["--oversubscribe"]
+    run_command += ["-n", str(nproc), os.path.join(artemis_exe_dir, "artemis")]
     if restart is not None:
-        run_command += ["-r", restart]
-    input_filename_full = os.path.join(artemis_inputs_dir, input_filename)
-    run_command += ["-i", input_filename_full]
+        run_command += ["-r", os.path.join(artemis_data_dir, restart)]
+    run_command += ["-i", os.path.join(artemis_inputs_dir, input_filename)]
+    run_command += ["-d", artemis_data_dir]
 
     try:
-        # os.chdir(run_directory)
-        os.chdir(artemis_run_dir)
+        os.chdir(artemis_exe_dir)
```

Comment:
I would prefer to keep the run directory as it was and instead copy `librebound.so` from the exe directory to the run directory. I assume things like core dumps would get placed in the exe directory in this model?

I would also prefer to not use `-d` and have the snapshots in the run directory.

Reply:
I believe the current design does everything the earlier iteration did (plus some edge case fixes) without a manual copy of `librebound.so`.  

I like that this design gives `parthenon` the control of routing outputs using its `-d` flag.  

Also, I believe the core dump files belong wherever the executable is.  

You can routed the snapshots wherever you like with `--output_dir` in the current design.

I have also printed every path possible (there are too many...) so that you can clearly see where everything goes.

Reply:
Sure, but at the expense of running in the executable directory (and needing to use `-d`), which I was trying hard to avoid doing in the last iteration of this. The reason to not do that is if someone wants to run the tests on a released executable. They should not be running inside that shared directory -- especially if there are going to be things written to that directory.

Reply:
Even `main` defaults to outputting to `exe_path`:

https://github.com/lanl/artemis/blob/develop/tst/run_tests.py#L307

I am open to changing this to `cwd`

Reply:
Deprecated by previous discussion